index.bs

```diff
@@ -795,6 +728,584 @@ the following list:
+
+   <p class=note>This means that backpressure will be applied when the user agent's buffers are
+   full.
+
+  1. If [=the WebSocket closing handshake is started|the closing handshake has not yet started=],
+     [=Send a WebSocket Message=] to |stream| comprised of |data| using |opcode|.
+  1. [=Queue a task=] on the [=WebSocket task source=] to resolve |promise| with undefined.
+ 1. Return |promise|.
+</div>
+
+<hr>
+
+<div algorithm="close algorithm">
+
+To <dfn>close</dfn> a {{WebSocketStream}} |stream|, [=close the WebSocket=] with |stream| (<var
+ignore>code</var> and <var ignore>reason</var> are not set).
+</div>
+
+<hr>
+
+<div algorithm="abort algorithm">
+
+To <dfn>abort</dfn> a {{WebSocketStream}} |stream| given |reason|, [=close using reason=] giving
+|stream| and |reason|.
+</div>
+
+<hr>
+
+<div algorithm="close using reason algorithm">
+
+To <dfn>close using reason</dfn> a {{WebSocketStream}} |stream| given |reason|:
+
+ 1. Let |code| be null.
+ 1. Let |reasonString| be the empty string.
+ 1. If |reason| [=implements=] {{WebSocketError}},
+  1. Set |code| to |reason|'s [=WebSocketError/closeCode=].
+  1. Set |reasonString| to |reason|'s [=WebSocketError/reason=].
+ 1. [=Close the WebSocket=] with |stream|, |code|, and |reasonString|. If this throws an exception,
+  discard |code| and |reasonString| and [=close the WebSocket=] with |stream|.
+
+ <p class=note>A {{WebSocketError}} object constructed from JavaScript will always have a
+ [=WebSocketError/closeCode=] and [=WebSocketError/reason=] that JavaScript is permitted to set.
+ However, the {{WebSocketStream/closed}} promise might be rejected with a {{WebSocketError}} whose
+ [=WebSocketError/closeCode=] has a value coming from the server that JavaScript is not permitted to
+ set itself, such as 1001 "Going Away".
+</div>
+
+
+
+## The {{WebSocketError}} interface ## {#the-websocketevent-interface}
+
+<dfn interface>WebSocketError</dfn> is a subclass of {{DOMException}}
+that represents the information associated with closing a WebSocket.
+
+<xmp class="idl">
+[Exposed=(Window,Worker)]
+interface WebSocketError : DOMException {
+  constructor(optional DOMString message = "",
+              optional WebSocketCloseInfo init = {});
+
+  readonly attribute unsigned short? closeCode;
+  readonly attribute USVString reason;
+};
+</xmp>
+
+Each {{WebSocketError}} object has an associated <dfn for="WebSocketError">closeCode</dfn>, which is
+a number.
+
+Each {{WebSocketError}} object has an associated <dfn for="WebSocketError">reason</dfn>, which is a
+string.
+
+<dl class="domintro non-normative">
+ : <code>|error| = new {{WebSocketError/constructor(message, init)|WebSocketError}}([|message| [,
+   |init| ] ]</code>
+ :: Creates a new {{WebSocketError}} object.
+
+  |message| is a string which will be used to initialize the {{DOMException/message}} attribute of
+  the base class.
+
+  The |init| argument is an object whose properties can be set as follows:
```

Comment:
Optional: you could instead spell out each overload, or inline the options values. Our style here is still settling down, and what you have here is also very acceptable.

Examples of various styles:

- https://html.spec.whatwg.org/#dom-navigation-forward-dev
- https://html.spec.whatwg.org/#dom-navigateevent-intercept-dev
- https://html.spec.whatwg.org/#dom-slot-assignedelements-dev
- https://html.spec.whatwg.org/#dom-navigation-navigate-dev

Reply:
I will stick with what I have for now, since it is working.